### PR TITLE
feat(LERN-47): Implemented Teacher and Student count endpoint.

### PR DESCRIPTION
**Pull request by SleepyPeel**

### CODE REVIEW

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserQueries.kt [hunk 1]
```diff
@@ -4,10 +4,32 @@ import ch.learntrack.backend.persistence.UserRole
...
+import java.util.UUID
 
 public fun UserDao.fetchAllAdminUsers(): List<User> = ctx()
     .select()
     .from(USER)
     .where(USER.USER_ROLE.eq(UserRole.ADMIN))
     .fetch()
     .into(User::class.java)
+
+public fun UserDao.fetchAllAdminsForSchool(schoolId: UUID): MutableList<User> = ctx()
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserQueries.kt:17]:
```suggestion
public fun UserDao.fetchAllAdminUserBySchoolId(schoolId: UUID): MutableList<User> = ctx()
```

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserQueries.kt [hunk 2]
```diff
@@ -4,10 +4,32 @@ import ch.learntrack.backend.persistence.UserRole
...
+    .select(USER)
+    .from(USER_SCHOOL)
+    .join(USER)
+    .on(USER.ID.eq(USER_SCHOOL.USER_ID))
+    .and(USER_SCHOOL.SCHOOL_ID.eq(schoolId))
+    .where(USER.USER_ROLE.eq(UserRole.ADMIN))
+    .fetch()
+    .into(User::class.java)
+
+public fun UserDao.countTeachersInSchool(schoolId: UUID, userRole: UserRole): Int = ctx()
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserQueries.kt:27]:
```suggestion
public fun UserDao.fetchUserCountBySchoolIdAndUserRole(schoolId: UUID, userRole: UserRole): Int = ctx()
```

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserResource.kt [hunk 3]
```diff
@@ -16,4 +21,18 @@ public class UserResource(private val userService: UserService) {
     @PostMapping("/create")
     public fun createAdminUser(@RequestBody createUserDto: CreateUserDto): Unit =
         userService.createAdminUser(createUserDto)
+
+    @GetMapping("/getAllAdminsForSchool")
+    public fun getAllAdminsForSchool(@RequestParam schoolId: UUID): List<UserDto> = userService.getAllAdminsForSchool(
+        schoolId,
+    )
+
+    //please someone who is smarter than me (me=Shay) fix this
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserResource.kt:30]: ??

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserResource.kt [hunk 4]
```diff
@@ -16,4 +21,18 @@ public class UserResource(private val userService: UserService) {
...
+    @GetMapping("/getTeacherCounts")
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserResource.kt:31]:
```suggestion
    @GetMapping("/getTeacherCount")
```
at least still don't like it but its okay for the moment.

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserResource.kt [hunk 5]
```diff
@@ -16,4 +21,18 @@ public class UserResource(private val userService: UserService) {
...
+    public fun getTeacherCounts(@RequestParam schoolId: UUID): Int = userService.getUserCount(schoolId,
+        UserRole.TEACHER)
+
+    @GetMapping("/getStudentCounts")
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserResource.kt:35]:
```suggestion
    @GetMapping("/getStudentCount")
```

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserService.kt [hunk 6]
```diff
@@ -30,6 +30,12 @@ public class UserService(
 
     public fun getAllAdminUsers(): List<UserDto> = userDao.fetchAllAdminUsers().map(::mapToDto)
 
+    public fun getAllAdminsForSchool(schoolId: UUID): List<UserDto> = userDao.fetchAllAdminsForSchool(schoolId).map(
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserService.kt:33]:
```suggestion
    public fun getAllAdminsBySchoolId(schoolId: UUID): List<UserDto> = userDao.fetchAllAdminsForSchool(schoolId).map(
```

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

**Review by deepspace1000 — CHANGES_REQUESTED** [review]
Write a test for the getAllAdmin user endpoint

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

File: backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserQueries.kt [hunk 7]
```diff
@@ -4,10 +4,32 @@ import ch.learntrack.backend.persistence.UserRole
...
+import java.util.UUID
 
 public fun UserDao.fetchAllAdminUsers(): List<User> = ctx()
     .select()
     .from(USER)
     .where(USER.USER_ROLE.eq(UserRole.ADMIN))
     .fetch()
     .into(User::class.java)
+
+public fun UserDao.fetchAllAdminUserBySchoolId(schoolId: UUID): MutableList<User> = ctx()
```
Comment (deepspace1000) [inline backend/src/main/kotlin/ch/learntrack/backend/backoffice/user/UserQueries.kt:17]:
```suggestion
public fun UserDao.fetchAllAdminUserBySchoolId(schoolId: UUID): List<User> = ctx()
```

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 1  | ❗       |
    


*❗ custom/admin-landing-page*: 


| Before | After |
| --- | --- |
|  |  |

**Review by deepspace1000 — APPROVED** [review]

**Pull request merged**